lib/compile.js

```diff
@@ -70,11 +97,20 @@ function compileOutputs(loader, tree, opts, sfxEntryPoints) {
       entry.loads.push(load);
     }
 
-    return Promise.resolve(compileLoad(loader, load, opts, compilers))
-    .then(function(output) {
-      outputs.push(output);
-    });
+    var output = retrieveCachedOutput(cache, load);
+    if (output)
+      return output;
+
+    return compileLoad(loader, load, opts, compilers).then(function(output) {
+      updateCacheEntry(cache, load, output);
+      return output;
+    })
   }))
+  .then(function(loadOutputs) {
+    outputs = loadOutputs.filter(function(output) {
```

Comment:
Happy to revert back to pushing to `outputs` if you prefer it though, would be a little bit more optimal just feels less "functional", probably a matter of taste, I'm big into Haskell/lodash which possibly affects my js pragmatism.